setup.sh

```diff
@@ -142,71 +165,55 @@ function update() {
     function update_project() {
         local project=$1
         local gitproject="git://github.com/dlang/$project.git"
-        if ! ( cd "$wd/$project" && \
-            git checkout master && \
-            git pull --ff-only $gitproject master && \
-            git pull $gitproject master --tags && \
-            git fetch $gitproject && \
-            git fetch --tags $gitproject) 2>$tempdir/$project.log
+        local git="git -C $wd/$project"
+        if ! ( \
+            $git checkout master && \
+            $git pull --ff-only --tags "$gitproject" master ) 2> "$tempdir/$project.log"
         then
-            echo "Failure updating $wd/$project." >>$tempdir/errors
+            echo "Failure updating $wd/$project." >> "$tempdir/errors"
             exit 1
         fi
     }
 
-    for project in $toUpdate; do
-        update_project $project &
+    for project in "${toUpdate[@]}" ; do
+        update_project "$project" &
     done
     wait
 
-    if [ -f $tempdir/errors ]; then
-        cat $tempdir/*.log >&2
+    if [ -f "$tempdir/errors" ]; then
+        cat "$tempdir"/*.log >&2
         exit 1
     fi
 }
 
 function makeWorld() {
-# First make dmd
-    (
-        which dmd >/dev/null || BT="AUTO_BOOTSTRAP=1"
-        cd "$wd/dmd/src" &&
-        $makecmd -f posix.mak clean MODEL=$model $BT &&
-        $makecmd -f posix.mak -j $parallel MODEL=$model $BT
-    )
-
-# Update the running dmd version
-    if [[ ! -z $install ]]; then
-        local old=$(which dmd)
+    local BOOTSTRAP=""
+    which dmd >/dev/null || BOOTSTRAP="AUTO_BOOTSTRAP=1"
+    for repo in dmd druntime phobos ; do
+        "$makecmd" -C "$wd/$repo" -f posix.mak clean
+        "$makecmd" -C "$wd/$repo" -f posix.mak "-j${parallel}" MODEL="$model" BUILD="$build" $BOOTSTRAP
+    done
+
+    # Update the running dmd version (only required once)
+    if [[ ! -z "${install+x}" ]]; then
+        local old dmdBinary
+        old=$(which dmd)
+        dmdBinary=$(ls -1 $wd/dmd/generated/*/$build/$model/dmd)
         if [ -f "$old" ]; then
-            echo "Copying "$wd/dmd/src/dmd" over $old"
-            [ ! -w "$old" ] && local sudo="sudo"
-            $sudo cp "$wd/dmd/src/dmd" "$old"
+            echo "Linking '$dmdBinary' to $old"
+            local sudo=""
+            if [ ! -w "$old" ] ; then
+                sudo="sudo"
+            fi
+            ln -s "$tempdir/dmd.symlink" "$old"
+            "$sudo" mv "$tempdir/dmd.symlink" "$old"
```

Comment:
Only symlinks will use the automatically generated `dmd.conf` and thus the freshly built druntime + phobos libraries.